### PR TITLE
Removing Schema.getGlobalDescribe() call and slight cleanup

### PR DESCRIPTION
**Pull request by jamessimone**
I'd also _love_ to introduce an enum for the `EQUAL_TO` .... `GREATER_THAN_OR_EQUAL_TO` stuff (AND WE CAN USE `Enum.valueOf()` now!!!!)

### CODE REVIEW

File: apex-validation-rules/main/default/classes/RecordValidator.cls [hunk 1]
```diff
@@ -591,13 +589,12 @@ public without sharing class RecordValidator {
             SObject parentRecord = record;
 
             for (Schema.SObjectField field : this.fieldChain) {
-                Schema.DescribeFieldResult fieldDescribe = field.getDescribe();
-                String relationshipName = fieldDescribe.getRelationshipName();
+                Schema.DescribeFieldResult localFieldDescribe = field.getDescribe();
 
-                if (fieldDescribe.getSoapType() != Schema.SoapType.Id) {
-                    return parentRecord.get(fieldDescribe.getName());
+                if (localFieldDescribe.getSoapType() != Schema.SoapType.Id) {
```
Comment (jamessimone) [inline apex-validation-rules/main/default/classes/RecordValidator.cls:594]: this is always a refactor that `apex-assist` suggests when local variables shadow instance variables

File: apex-validation-rules/main/default/classes/RecordValidator.cls [hunk 2]
```diff
@@ -625,7 +622,7 @@ public without sharing class RecordValidator {
                 if (i == lastFieldIndex) {
                     fieldApiName = fieldPathPiece;
                 } else if (fieldPathPiece.endsWith('__r')) {
-                    fieldApiName = fieldPathPiece.replace('__r', '__c');
+                    fieldApiName = fieldPathPiece.removeEnd('__r') + '__c';
```
Comment (jongpie) [inline apex-validation-rules/main/default/classes/RecordValidator.cls:625]: ....omg, this approach is so much better!

File: apex-validation-rules/main/default/classes/RecordValidator.cls [hunk 3]
```diff
@@ -574,11 +574,9 @@ public without sharing class RecordValidator {
         private Schema.SObjectType sobjectType;
 
         public FieldPath(String sobjectApiName, String fieldPath) {
-            Schema.SObjectType sobjectType = Schema.getGlobalDescribe().get(sobjectApiName);
-
-            this.fieldChain = this.getFieldChain(sobjectType, fieldPath);
+            this.sobjectType = ((SObject) Type.forName(sobjectApiName).newInstance()).getSObjectType();
```
Comment (jongpie) [inline apex-validation-rules/main/default/classes/RecordValidator.cls:577]: Awww yeah, get out of here, global describe!

**Review by jongpie — APPROVED** [review]
I love it - thanks for making these changes! Feel free to merge it

File: apex-validation-rules/main/default/classes/RecordValidator.cls (hunk 2 above)
Comment (jamessimone) [reply apex-validation-rules/main/default/classes/RecordValidator.cls:625]: I learned it from you!

Comment (jongpie) [reply apex-validation-rules/main/default/classes/RecordValidator.cls:625]: ...oh! Well, this is awkward... :laughing:

**Pull request merged**